CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFile.swift

```diff
@@ -42,7 +42,19 @@ final class CEWorkspaceFile: Codable, Comparable, Hashable, Identifiable, Editor
     var name: String { url.lastPathComponent.trimmingCharacters(in: .whitespacesAndNewlines) }
 
     /// Returns the extension of the file or an empty string if no extension is present.
-    var type: FileIcon.FileType { .init(rawValue: url.pathExtension) ?? .txt }
+    var type: FileIcon.FileType {
+        let fileName = url.lastPathComponent
+        let fileExtension = url.pathExtension
+
+        // Verifies if the filename consists solely file extensions, such as '.env', '.gitignore', etc.
+        if fileName.hasPrefix(".") {
```

Comment:
This is not very efficient in my opinion...  Instead of using `hasPrefix(".")`, we check `fileName.first == "."`. This is faster as it only checks the first character.

Reply:
[`hasPrefix`](https://github.com/swiftlang/swift/blob/d443fdd501402444689c14c268e7052e85442097/stdlib/public/core/StringLegacy.swift#L124) will just grab a slice of the string equal to the length of the passed string and do a compare on it. In this case there's no difference in performance.